src/Microsoft.ML.TimeSeries/RootCauseLocalizationType.cs

```diff
@@ -0,0 +1,308 @@
+            Slices = slices;
+            AggType = aggregateType;
+            AggSymbol = aggregateSymbol;
+        }
+
+        public RootCauseLocalizationInput(DateTime anomalyTimestamp, Dictionary<string, Object> anomalyDimension, List<MetricSlice> slices, string aggregateSymbol)
+        {
+            AnomalyTimestamp = anomalyTimestamp;
+            AnomalyDimension = anomalyDimension;
+            Slices = slices;
+            AggType = AggregateType.Unknown;
+            AggSymbol = aggregateSymbol;
+        }
+    }
+
+    public sealed class RootCauseDataViewType : StructuredDataViewType
+    {
+        public RootCauseDataViewType()
+           : base(typeof(RootCause))
+        {
+        }
+
+        public override bool Equals(DataViewType other)
+        {
+            if (other == this)
+                return true;
+            if (!(other is RootCauseDataViewType tmp))
+                return false;
+            return true;
+        }
+
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        public override string ToString()
+        {
+            return typeof(RootCauseDataViewType).Name;
+        }
+    }
+
+    public sealed class RootCauseLocalizationInputDataViewType : StructuredDataViewType
+    {
+        public RootCauseLocalizationInputDataViewType()
+           : base(typeof(RootCauseLocalizationInput))
+        {
+        }
+
+        public override bool Equals(DataViewType other)
+        {
+            if (!(other is RootCauseLocalizationInputDataViewType tmp))
+                return false;
+            return true;
+        }
+
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        public override string ToString()
+        {
+            return typeof(RootCauseLocalizationInputDataViewType).Name;
+        }
+    }
+
+    public enum AggregateType
+    {
+        /// <summary>
+        /// Make the aggregate type as sum.
+        /// </summary>
+        Unknown = 0,
+        /// <summary>
+        /// Make the aggregate type as sum.
+        /// </summary>
+        Sum = 1,
+        /// <summary>
+        /// Make the aggregate type as average.
+        ///  </summary>
+        Avg = 2,
+        /// <summary>
+        /// Make the aggregate type as min.
+        /// </summary>
+        Min = 3,
+        /// <summary>
+        /// Make the aggregate type as max.
+        /// </summary>
+        Max = 4,
+        /// <summary>
+        /// Make the aggregate type as count.
+        /// </summary>
+        Count = 5
+    }
+
+    public enum AnomalyDirection
+    {
+        /// <summary>
+        /// the value is larger than expected value.
+        /// </summary>
+        Up = 0,
+        /// <summary>
+        /// the value is lower than expected value.
+        ///  </summary>
+        Down = 1
+    }
+
+    public sealed class RootCauseItem : IEquatable<RootCauseItem>
+    {
+        public double Score;
```

Comment:
Can you explain what the score is? (Also, is it ML.NET style to have public member variables?)

Reply:
> Can you explain what the score is? (Also, is it ML.NET style to have public member variables?)

Explaination is added. For the public member, @harishsk , could you help give some guidance on this? I saw both public and private members in the code?

Reply:
If the class and its members need to be part of the public API then it is public. Otherwise, it should private or internal.

---
In reply to: [421321079](https://github.com/dotnet/machinelearning/pull/4925#discussion_r421321079) [](ancestors = 421321079)

Reply:
> If the class and its members need to be part of the public API then it is public. Otherwise, it should be private or internal.
> If it is going to be public, can you please annotate the comments between ///
> 
> and ///
> like above
> In reply to: [421321079](https://github.com/dotnet/machinelearning/pull/4925#discussion_r421321079) [](ancestors = 421321079)

Have updated the comments inside ///  and ///